vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -72,40 +74,53 @@ class TextShadowNode final : public ShadowNodeBase {
     Super::AddView(child, index);
 
     if (auto run = static_cast<ShadowNodeBase &>(child).GetView().try_as<winrt::Run>()) {
-      if (m_ColorValue) {
-        AddHighlighter(m_ColorValue.value(), run.Text().size());
+      if (m_backgroundColor) {
+        AddHighlighter(m_backgroundColor, m_foregroundColor, run.Text().size());
       }
-
       m_prevCursorEnd += run.Text().size();
     } else if (auto span = static_cast<ShadowNodeBase &>(child).GetView().try_as<winrt::Span>()) {
-      AddNestedTextHighlighter(m_ColorValue, span, static_cast<VirtualTextShadowNode &>(child).m_highlightData);
+      AddNestedTextHighlighter(
+          m_backgroundColor, m_foregroundColor, span, static_cast<VirtualTextShadowNode &>(child).m_highlightData);
     }
   }
 
   void AddNestedTextHighlighter(
       const std::optional<winrt::Windows::UI::Color> &parentColor,
+      const std::optional<winrt::Windows::UI::Color> &parentForeColor,
```

Comment:
nit: consider renaming the "parentColor" param to make it explicit it is the background color.